src/components/new-user-form/index.tsx

```diff
@@ -0,0 +1,211 @@
+import { ApolloError } from "@apollo/client";
+import React, { FormEvent, useState } from "react";
+import { formatPhone } from "../../utils/phone-formatter";
+import { validateBirthDate } from "../../validators/birthdate";
+import { validateConfirmPassword } from "../../validators/confirm-password";
+import { validateEmail } from "../../validators/email";
+import { validateName } from "../../validators/name";
+import { validatePassword } from "../../validators/password";
+import { validatePhone } from "../../validators/phone";
+import { Button } from "../button";
+import { Input } from "../input";
+import * as S from "./style";
+
+interface NewUserParamsTypes {
+  variables: {
+    data: {
+      name: string;
+      email: string;
+      phone: string;
+      birthDate: string;
+      password: string;
+      role: string;
+    };
+  };
+}
+
+interface NewUserFormTypes {
+  createUserFunction: (data: NewUserParamsTypes) => void;
```

Comment:
como aqui é um evento, o que acha?
```suggestion
  onSubmitSuccess: (formData: NewUserFormData) => void;
```

Reply:
Muito melhor, e facilita demais o entendimento. Valeu demais Beto!